drivers/dose/dose.c

```diff
@@ -107,11 +111,112 @@ static inline void _disable_sense(dose_t *ctx)
 #endif
 }
 
+#ifdef MODULE_DOSE_WATCHDOG
+static unsigned _watchdog_users;
+static dose_t *_dose_base;
+static uint8_t _dose_numof;
+
+#ifndef DOSE_TIMER_DEV
+static xtimer_t _wdt;
+static unsigned _wdt_period;
+#endif
+
+static inline void _watchdog_start(void)
+{
+    if (_watchdog_users) {
+        return;
+    }
+
+    _watchdog_users++;
+#ifdef DOSE_TIMER_DEV
+    timer_start(DOSE_TIMER_DEV);
+#else
+    xtimer_set(&_wdt, _wdt_period);
```

Comment:
The software timer fallback doesn't improve anything compared to not using `dose_watchdog`, but we could use it to eliminate the need for a `dose_watchdog` pseudo-module and always enable the watchdog code.

Reply:
If it doesn't improve anything could we remove it?